tests/admin/test_controller.py

```diff
@@ -1081,6 +1112,236 @@ def test_classifications(self):
+                ("rights_explanation", "explanation"),
+            ])
+            flask.request.files = MultiDict([
+                ("cover_file", TestFileUpload(image_data)),
+            ])
+            response = self.manager.admin_work_controller.change_book_cover(identifier.type, identifier.identifier, mirror)
+            eq_(200, response.status_code)
+
+            [link] = identifier.links
+            eq_(Hyperlink.IMAGE, link.rel)
+            eq_(staff_data_source, link.data_source)
+
+            resource = link.resource
+            assert identifier.urn in resource.url
+            eq_(staff_data_source, resource.data_source)
+            eq_(RightsStatus.CC_BY, resource.rights_status.uri)
+            eq_("explanation", resource.rights_explanation)
+
+            representation = resource.representation
+            [thumbnail] = resource.representation.thumbnails
+
+            eq_(resource.url, representation.url)
+            eq_(Representation.PNG_MEDIA_TYPE, representation.media_type)
+            eq_(Representation.PNG_MEDIA_TYPE, thumbnail.media_type)
+            eq_(image_data, representation.content)
+            assert identifier.identifier in representation.mirror_url
+            assert identifier.identifier in thumbnail.mirror_url
+
+            eq_([], process_called_with)
+            eq_([representation, thumbnail], mirror.uploaded)
+            eq_([representation.mirror_url, thumbnail.mirror_url], mirror.destinations)
+
+        work = self._work(with_license_pool=True)
+        identifier = work.license_pools[0].identifier
+
+        # Upload a new cover image and add the title and author to it.
+        # Both the original image and the generated image will become resources.
+        with self.request_context_with_library_and_admin("/"):
+            flask.request.form = MultiDict([
+                ("title_position", "center"),
+                ("rights_status", RightsStatus.CC_BY),
+                ("rights_explanation", "explanation"),
+            ])
+            flask.request.files = MultiDict([
+                ("cover_file", TestFileUpload(image_data)),
+            ])
+            response = self.manager.admin_work_controller.change_book_cover(identifier.type, identifier.identifier, mirror)
+            eq_(200, response.status_code)
+
+            [link] = identifier.links
+            eq_(Hyperlink.IMAGE, link.rel)
+            eq_(staff_data_source, link.data_source)
+
+            resource = link.resource
+            assert identifier.urn in resource.url
+            eq_(staff_data_source, resource.data_source)
+            eq_(RightsStatus.CC_BY, resource.rights_status.uri)
+            eq_("explanation", resource.rights_explanation)
+
+            derivation = self._db.query(ResourceDerivation).filter(ResourceDerivation.derivative_id==resource.id).one()
+            original_resource = derivation.original
+            assert resource != original_resource
```

Comment:
Can you also verify that this resource's thumbnail is different from the old resource's thumbnail?